src/main.js

```diff
@@ -1,144 +1,175 @@
 'use babel'
 
-// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
-import { CompositeDisposable } from 'atom'
-import { hasValidScope } from './validate/editor'
-import {
-  atomConfig,
-  jobConfig,
-  getMigrations,
-  subscribe as configSubscribe
-} from './atom-config'
-
-// Internal variables
-const idleCallbacks = new Set()
-
 // Dependencies
-// NOTE: We are not directly requiring these in order to reduce the time it
-// takes to require this file as that causes delays in Atom loading this package
-let path
-let worker
-let configInspector
-let debug
-let linterMessage
-let knownRules
-
-const loadDeps = () => {
-  if (!path) {
-    path = require('path')
-  }
-  if (!worker) {
-    worker = require('./worker-manager')
-  }
-  if (!configInspector) {
-    configInspector = require('./eslint-config-inspector')
-  }
-  if (!debug) {
-    debug = require('./debug')
-  }
-  if (!linterMessage) {
-    linterMessage = require('./linter-message')
-  }
-  if (!knownRules) {
-    knownRules = require('./rules').default
-  }
-}
+// NOTE: Requiring files adds significant time to package startup. To avoid this
+// delay, we are *not* requiring at the top of this file. Use idleCallbacks to
+// attempt to pre-cache requires for fast resolution if scheduler allows. Since
+// all requires are statically defined and inside some function, Atom should be
+// able "snapshot" the package.
+//
+const dependencies = [
+  () => require('./validate/editor'),
+  () => require('./linter-message'),
+  () => require('./worker-manager'),
+  () => require('./eslint-config-inspector'),
+  () => require('./debug'),
+  () => require('./rules'),
+  () => require('path'),
+  () => require('./atom-config'),
+]
+
+// Miscellaneous idle tasks.
+//
+const idleTasks = [
```

Comment:
This is a static list that is only ever used in one place, just embed it there.

Reply:
Sorry, I don't understand. Could you rephrase?

Reply:
Oh, I think I get it after seeing you next same request. This is similar to listing imports at the top of the file. Even though we are not actually _doing_ the import, it still gives an overview of what's coming in. Additionally it separates data from behavior. It' much easier to comprehend the data (a list of dependencies) when not mixed in with behavior (lazily attempting to import files), and it is equally easier to comprehend the behavior when not having to parse through a list of files.

Reply:
> Even though we are not actually doing the import, it still gives an overview of what's coming in.

You're abusing `require` all through the rest of the file, so this no longer has any meaning.

----

I'm just saying move this (and the next) block down to right before it is actually used. That way it is scoped into the `activate` and gotten rid of when no longer used, as well as having the advantage that you don't have to go searching for what  those lines are actually doing when you see them.

Reply:
Not a fan of what that does to the visual organization of the code, but you make a fair point. My plan is to concede this one without further argument.